python/ray/air/execution/resources/resource_manager.py

```diff
@@ -0,0 +1,154 @@
+import abc
+
+from typing import List, Optional
+
+import ray
+from ray.air.execution.resources.request import (
+    ResourceRequest,
+    AcquiredResource,
+)
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class ResourceManager(abc.ABC):
+    """Resource manager interface.
+
+    A resource manager can be used to request resources from a Ray cluster and
+    allocate them to remote Ray tasks.
+
+    Resources have to be requested before they can be acquired.
+
+    Resources managed by the resource manager can be in three states:
+
+    1. "Requested":  The resources have been requested but are not yet available to
+       schedule remote Ray objects. The resource request may trigger autoscaling,
+       and can be cancelled if no longer needed.
+    2. "Ready": The requested resources are now available to schedule remote Ray
+       objects. They can be acquired and subsequently used remote Ray objects.
+       The resource request can still be cancelled if no longer needed.
+    3. "Acquired": The resources have been acquired by a caller to use for scheduling
+       remote Ray objects. Note that it is the responsibility of the caller to
+       schedule the Ray objects with these resources.
+       The associated resource request has been removed and can no longer be cancelled.
```

Comment:
nit: I feel like "removed" isn't very clear (removed from where?)
```suggestion
       The associated resource request has been completed and can no longer be cancelled.
```